### PR TITLE
Feat: Support Tiered Margins

### PR DESCRIPTION
**Pull request by paulofph**

### CODE REVIEW

File: src/utils/index.ts [hunk 1]
```diff
@@ -351,21 +352,54 @@ export const computeExternalGetAGPriceItemValues = (
...
+          unitAmountNet: isTaxInclusive
+            ? toDinero(getAg.markup_amount_decimal)
+                .divide(1 + taxRate)
+                .getAmount()
+            : toDinero(getAg.markup_amount_decimal).getAmount(),
+        } as PriceItemsTotals);
+
+  const relevantTier = markupValues.tiers_details?.[0];
+  const unitAmountGetAgFeeNet =
+    getAg.type === 'base_price'
```
Comment (alexmarqs) [inline src/utils/index.ts:387]: Can we just use a fallback mechanism in case we dont have type? If so, use the variable_price to undertand

Comment (paulofph) [reply src/utils/index.ts:388]: It doesn't make sense anymore. Because base_price can be or not to be variable.

Comment (paulofph) [reply src/utils/index.ts]: As aligned before, let's migrated the old prices instead

File: src/pricing.ts [hunk 2]
```diff
@@ -50,6 +50,15 @@ export enum PricingModel {
   tieredFlatFee = 'tiered_flatfee',
   externalGetAG = 'external_getag',
 }
+export enum MarkupPricingModel {
```
Comment (alexmarqs) [inline src/pricing.ts:53]: we have this now exported from the client right?

Comment (alexmarqs) [reply src/pricing.ts:53]: I think I know why, it's because its not a enum, got it

**Pull request merged**